external-crates/move/crates/move-binary-format/src/normalized.rs

```diff
@@ -1,24 +1,40 @@
-// Copyright (c) The Diem Core Contributors
 // Copyright (c) The Move Contributors
-// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-S: Apache-2.0
 
 use crate::file_format::{
-    AbilitySet, Bytecode as FBytecode, CodeOffset, CompiledModule, DatatypeTyParameter,
-    EnumDefinition, FieldDefinition, FieldHandle, FieldHandleIndex, FieldInstantiation,
-    FieldInstantiationIndex, FunctionDefinition, FunctionHandle, FunctionHandleIndex,
-    FunctionInstantiation, JumpTableInner, LocalIndex, SignatureIndex, SignatureToken,
-    StructDefInstantiation, StructDefInstantiationIndex, StructDefinition, StructDefinitionIndex,
-    StructFieldInformation, TypeParameterIndex, VariantDefinition, VariantHandleIndex,
-    VariantInstantiationHandleIndex, VariantJumpTable as FFVariantJumpTable, Visibility,
+    self, AbilitySet, Bytecode as FBytecode, CodeOffset, CompiledModule, DatatypeHandleIndex,
+    DatatypeTyParameter, EnumDefinition, EnumDefinitionIndex, FieldDefinition, FieldHandleIndex,
+    FieldInstantiationIndex, FunctionDefinition, FunctionHandleIndex, FunctionInstantiationIndex,
+    JumpTableInner, LocalIndex, SignatureIndex, SignatureToken, StructDefInstantiationIndex,
+    StructDefinition, StructDefinitionIndex, StructFieldInformation, TypeParameterIndex,
+    VariantDefinition, VariantHandleIndex, VariantInstantiationHandleIndex, VariantTag, Visibility,
 };
 use move_core_types::{
     account_address::AccountAddress,
     identifier::{IdentStr, Identifier},
-    language_storage::{ModuleId, StructTag, TypeTag},
+    language_storage::{StructTag, TypeTag},
 };
-use move_proc_macros::test_variant_order;
 use serde::{Deserialize, Serialize};
-use std::collections::BTreeMap;
+use std::{
+    borrow::Borrow,
+    collections::{BTreeMap, BTreeSet},
+};
+use std::{fmt::Debug, ops::Deref};
+use std::{rc::Rc, sync::Arc};
+
+pub trait StringPool {
+    type String;
+
+    fn intern(&mut self, s: &IdentStr) -> Self::String;
+
+    fn as_ident_str<'a>(&'a self, s: &'a Self::String) -> &'a IdentStr;
+}
+
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
+pub struct ModuleId<S> {
```

Comment:
This name feels a bit confusing--open to suggestions

Reply:
I think this name is fine so no suggestions from me.